### PR TITLE
Fix update bug

### LINKED ISSUE #182: [PE-D][Tester C] Deletion of all data due to allowing of invalid formats to pass through
As mentioned in one of the previous bug reports (bug report 4), it seems that the update_member command allows for invalid formats. Hence, because of this, if I have an empty list and were to do `add_member n/John Doe r/4-3-301 t/johndoe123 tag/logistics` followed by `add_member n/Hi Doe r/4-3-301 t/hidoe123 tag/logistics`, and i do `update_member 2 n/John Doe`, this will result in 2 people having the same name as seen in **image 1**. With reference to the **Caution** in the segment of the User Guide about **Editing the data file**, because of this invalid data that is accepted to be written and saved into the data file, if you were to exit and open back up the application, you lose all the data previously stored as seen in **image 2** as a result of the actions taken when there is invalid data formats as mentioned in **Caution**.

**Image 1 (2 people with same name as a result of the sequence of commands mentioned above):**

**Image 2 (After exiting and reopening the application, all data lost):**

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: nws321/ped#5

### LINKED ISSUE #214: [PE-D][Tester B] There can be two people with the same telegram ID when using the update_member command
After I keyed in `update_member 1 t/betsy_crows` I ended up having two people with the same telegram id. However when I am adding another person with the same telegram ID, I am not able to add that person in (as shown in the screenshot below).

This could be due to the fact that only the `name` is being checked for the `update_member` command and not the `telegram` field.

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: SwaminathanViswa/ped#5

### LINKED ISSUE #229: Telegram case insensitity
telegram should be case insensitive

### PR DESCRIPTION
**Pull request by taggyhan**
Fixes Issues regarding updating member and telegram

 Fix Telegram case insensitivity
 Widen scope of otherwise incorrect error message

Fix #229 , #214

### CODE REVIEW

File: src/main/java/hallpointer/address/logic/commands/UpdateMemberCommand.java [hunk 1]
```diff
@@ -51,8 +51,8 @@ public class UpdateMemberCommand extends Command {
 
     public static final String MESSAGE_SUCCESS = "Member %1$s's details updated successfully.";
     public static final String MESSAGE_NOT_UPDATED = "At least one field to update must be provided.";
-    public static final String MESSAGE_DUPLICATE_MEMBER = "A member with the same name already exists.";
     public static final String MESSAGE_INVALID_INDEX = "Error: Invalid index specified.";
+    public static final String MESSAGE_DUPLICATE_MEMBER = "A member with the same name or telegram already exists.";
```
Comment (QinHaichen12) [inline src/main/java/hallpointer/address/logic/commands/UpdateMemberCommand.java:55]: UG guide update

File: src/main/java/hallpointer/address/model/member/Member.java [hunk 2]
```diff
@@ -167,14 +167,10 @@ public void removeSession(SessionName sessionName) {
...
     public boolean isSameMember(Member otherMember) {
-        if (otherMember == this) {
-            return true;
-        }
-
         // Check for duplicate based only on name and telegram
         return otherMember != null
                 && (otherMember.getName().equals(getName())
-                || otherMember.getTelegram().equals(getTelegram()));
+                || otherMember.getTelegram().value.equalsIgnoreCase(this.getTelegram().value));
```
Comment (QinHaichen12) [inline src/main/java/hallpointer/address/model/member/Member.java:173]: duplicate check now includes telegram, maybe can change telegram equals method

File: src/main/java/hallpointer/address/logic/commands/UpdateMemberCommand.java [hunk 3]
```diff
@@ -80,11 +87,16 @@ public CommandResult execute(Model model) throws CommandException {
...
         if (!memberToUpdate.isSameMember(updatedMember) && model.hasMember(updatedMember)) {
             throw new CommandException(MESSAGE_DUPLICATE_MEMBER);
         }
 
+        long matchCount = model.countMemberOccurrences(updatedMember);
+
+        if (memberToUpdate.isSameMember(updatedMember) && matchCount > 1) {
+            throw new CommandException(MESSAGE_DUPLICATE_MEMBER);
+        }
+
```
Comment (QinHaichen12) [inline src/main/java/hallpointer/address/logic/commands/UpdateMemberCommand.java:99]: for the case where at least one of name or tele is not altered

File: src/main/java/hallpointer/address/model/member/Member.java (hunk 2 above)
Comment (QinHaichen12) [reply src/main/java/hallpointer/address/model/member/Member.java:173]: equality check is no longer transitive

**Review by QinHaichen12 — COMMENTED** [review]
LGTM! Merged

File: src/main/java/hallpointer/address/model/member/Member.java (hunk 2 above)
Comment (taggyhan) [reply src/main/java/hallpointer/address/model/member/Member.java:173]: yes should

**Pull request merged**